lib/shopify_cli/theme/dev_server/hot-reload.js

```diff
@@ -1,93 +1,131 @@
 (() => {
-  function connect() {
-    const eventSource = new EventSource('/hot-reload');
-
-    eventSource.onmessage = handleUpdate;
-
-    eventSource.onopen = () => console.log('[HotReload] SSE connected.');
-
-    eventSource.onclose = () => {
-      console.log('[HotReload] SSE closed. Attempting to reconnect...');
-
-      setTimeout(connect, 5000);
+  class SectionReloadEvent extends Event {
+    constructor(filename, name, id) {
+      super('shopify:hot:section', { bubbles: true, cancelable: false });
+      this.filename = filename;
+      this.name = name;
+      this.id = id;
     }
-
-    eventSource.onerror = () => eventSource.close();
   }
 
-  connect();
-
-  function handleUpdate(message) {
-    var data = JSON.parse(message.data);
-
-    // Assume only one file is modified at a time
-    var modified = data.modified[0];
-
-    if (isCssFile(modified)) {
-      reloadCssFile(modified)
-    } else if (isSectionFile(modified)) {
-      reloadSection(modified);
-    } else {
-      console.log(`[HotReload] Refreshing entire page`);
-      window.location.reload();
+  class StylesheetReloadEvent extends Event {
+    constructor(filename, links) {
+      super('shopify:hot:stylesheet', { bubbles: true, cancelable: false });
```

Comment:
We're currently dispatching this event after changing the `link.href`, which triggers an async network request under the hoods.

So, following the convention mentioned above, I believe we could rename it to `shopify:stylesheet:hotreload`  (suggesting that it may not be loaded).